packages/@expo/cli/src/start/server/metro/inspector-proxy/proxy.ts

```diff
@@ -0,0 +1,155 @@
+    this.metroProxy.processRequest(req, res, next);
+  }
+
+  public createWebSocketListeners(server: HttpServer | HttpsServer): Record<string, WSServer> {
+    this.setServerAddress(server);
+
+    return {
+      [WS_DEVICE_URL]: this.createDeviceWebSocketServer(),
+      [WS_DEBUGGER_URL]: this.createDebuggerWebSocketServer(),
+    };
+  }
+
+  private createDeviceWebSocketServer() {
+    const wss = new WS.Server({
+      noServer: true,
+      perMessageDeflate: false,
+    });
+
+    // See: https://github.com/facebook/metro/blob/eeb211fdcfdcb9e7f8a51721bd0f48bc7d0d211f/packages/metro-inspector-proxy/src/InspectorProxy.js#L157
+    wss.on('connection', (socket, request) => {
+      try {
+        const deviceId = this.metroProxy._deviceCounter++;
+        const { deviceName, appName } = getNewDeviceInfo(request.url);
+
+        this.devices.set(
+          deviceId,
+          new this.DeviceClass(deviceId, deviceName, appName, socket, this.metroProxy._projectRoot)
+        );
+
+        debug('New device connected: device=%s, app=%s', deviceName, appName);
+
+        socket.on('close', () => {
+          this.devices.delete(deviceId);
+          debug('Device disconnected: device=%s, app=%s', deviceName, appName);
+        });
+      } catch (error: unknown) {
+        const message = error instanceof Error && error.toString();
+        console.error('Could not establish a connection to device:', error);
+        socket.close(WS_GENERIC_ERROR_STATUS, message || 'Unknown error');
+      }
+    });
+
+    return wss;
+  }
+
+  private createDebuggerWebSocketServer() {
+    const wss = new WS.Server({
+      noServer: true,
+      perMessageDeflate: false,
+    });
+
+    // See: https://github.com/facebook/metro/blob/eeb211fdcfdcb9e7f8a51721bd0f48bc7d0d211f/packages/metro-inspector-proxy/src/InspectorProxy.js#L193
+    wss.on('connection', (socket, request) => {
+      try {
+        const { deviceId, pageId } = getExistingDeviceInfo(request.url);
+        if (!deviceId || !pageId) {
+          throw new Error(`Missing "device" and/or "page" IDs in query parameters`);
+        }
+
+        const device = this.devices.get(parseInt(deviceId, 10));
+        if (!device) {
+          throw new Error(`Device with ID "${deviceId}" not found.`);
+        }
+
+        debug('New debugger connected: device=%s, app=%s', device._name, device._app);
+
+        device.handleDebuggerConnection(socket, pageId);
+
+        socket.on('close', () => {
+          debug('Debugger disconnected: device=%s, app=%s', device._name, device._app);
+        });
+      } catch (error: unknown) {
+        console.log('ERRR', error);
+        const message = error instanceof Error && error.toString();
+        debug('Could not establish a connection to debugger:', error);
+        socket.close(WS_GENERIC_ERROR_STATUS, message || 'Unknown error');
+      }
+    });
+
+    return wss;
+  }
+}
+
+function asString(value?: string | string[]) {
```

Comment:
add return type for conversion functions

Reply:
```suggestion
function asString(value?: string | string[]): string {
```